Mathlib/Geometry/Euclidean/Sphere/Tangent.lean

```diff
@@ -178,6 +188,49 @@ lemma IsTangentAt.isTangent {s : Sphere P} {p : P} {as : AffineSubspace ℝ P}
   · rw [center_mem_orthRadius_iff] at hsp
     rwa [← hsp] at hs
 
+lemma IsTangent.infDist_eq_radius {s : Sphere P} {as : AffineSubspace ℝ P} (h : s.IsTangent as) :
+    Metric.infDist s.center as = s.radius := by
+  obtain ⟨p, h⟩ := h
+  refine le_antisymm ?_ ?_
+  · convert Metric.infDist_le_dist_of_mem h.mem_space
+    rw [mem_sphere'.1 h.mem_sphere]
+  · rw [Metric.infDist_eq_iInf]
+    have : Nonempty as := ⟨⟨p, h.mem_space⟩⟩
+    refine le_ciInf fun x ↦ le_of_sq_le_sq ?_ dist_nonneg
+    rw [dist_comm, h.dist_sq_eq_of_mem x.property, le_add_iff_nonneg_right]
+    exact sq_nonneg _
+
+lemma dist_orthogonalProjection_eq_radius_iff_isTangentAt {s : Sphere P} {as : AffineSubspace ℝ P}
+    [Nonempty as] [as.direction.HasOrthogonalProjection] :
+    dist s.center (orthogonalProjection as s.center) = s.radius ↔
+      s.IsTangentAt (orthogonalProjection as s.center) as := by
+  refine ⟨fun h ↦ ?_, fun h ↦ ?_⟩
+  · refine ⟨?_, orthogonalProjection_mem _, fun p hp ↦ ?_⟩
+    · rwa [mem_sphere']
+    · rw [SetLike.mem_coe, mem_orthRadius_iff_inner_left]
+      exact orthogonalProjection_vsub_mem_direction_orthogonal as s.center _
+        (vsub_orthogonalProjection_mem_direction s.center hp)
+  · rw [dist_orthogonalProjection_eq_infDist, h.isTangent.infDist_eq_radius]
+
+lemma dist_orthogonalProjection_eq_radius_iff_isTangent {s : Sphere P} {as : AffineSubspace ℝ P}
+    [Nonempty as] [as.direction.HasOrthogonalProjection] :
+    dist s.center (orthogonalProjection as s.center) = s.radius ↔ s.IsTangent as := by
+  refine ⟨fun h ↦ (dist_orthogonalProjection_eq_radius_iff_isTangentAt.1 h).isTangent, fun h ↦ ?_⟩
+  rw [dist_orthogonalProjection_eq_infDist, h.infDist_eq_radius]
+
+lemma isTangent_iff_isTangentAt_orthogonalProjection {s : Sphere P} {as : AffineSubspace ℝ P}
```

Comment:
I think `alias \<IsTangent.isTangentAt, _> := isTangent_iff_isTangentAt_orthogonalProjection` would be sensible here.

Reply:
Added.